python/pyiceberg/avro/resolver.py

```diff
@@ -150,8 +152,12 @@ def struct(self, struct: StructType, expected_struct: Optional[IcebergType], fie
             if read_field.field_id not in file_fields:
                 if read_field.required:
                     raise ResolveError(f"{read_field} is non-optional, and not part of the file schema")
-                # Just set the new field to None
-                results.append((pos, NoneReader()))
```

Comment:
If the initial default is present, then the field _can_ be required. It is valid to add a new required field as long as it has a non-null initial default.

Reply:
Ah nice, I missed that one. Updated!